src/ApiManagement/ApiManagement.ServiceManagement/Commands/NewAzureApiManagementApiSchema.cs

```diff
@@ -0,0 +1,101 @@
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementApiSchema", SupportsShouldProcess = true, DefaultParameterSetName = SchemaDocumentInlineParameterSet)]
+    [OutputType(typeof(PsApiManagementApiSchema), ParameterSetName = new[] { SchemaDocumentInlineParameterSet, SchemaDocumentFromFileParameterSet })]
+    public class NewAzureApiManagementApiSchema : AzureApiManagementCmdletBase
+    {
+        #region ParameterSets
+        private const string SchemaDocumentInlineParameterSet = "SchemaDocumentInline";
+        private const string SchemaDocumentFromFileParameterSet = "SchemaDocumentFromFile";
+        #endregion
+
+        [Parameter(
+            ValueFromPipeline = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Identifier of api. This parameter is required.")]
+        public String ApiId { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Identifier of schema in the api. This parameter is optional. If not specified we will generate it.")]
+        public String SchemaId { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "ContentType of the api Schema. This parameter is required.")]
+        [PSArgumentCompleter("swaggerdefinition", "openapicomponents", "xsdschema", "wadlgrammar")]
+        public String SchemaDocumentContentType { get; set; }
+
+        [Parameter(
+            ParameterSetName = SchemaDocumentInlineParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Api schema document as a string. This parameter is required is -SchemaDocumentFile is not specified.")]
+        public String SchemaDocument { get; set; }
+
+        [Parameter(
+            ParameterSetName = SchemaDocumentFromFileParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Api schema document file path. This parameter is required is -SchemaDocument is not specified.")]
+        public String SchemaDocumentFilePath { get; set; }
+
+        public override void ExecuteApiManagementCmdlet()
+        {
+            string schemaID = SchemaId ?? Guid.NewGuid().ToString();
+            string schemaDocument;
+
+            if (ParameterSetName.Equals(SchemaDocumentInlineParameterSet))
+            {
+                schemaDocument = SchemaDocument;
+            }
+            else 
+            {
+                schemaDocument = File.ReadAllText(SchemaDocumentFilePath);                
```

Comment:
@solankisamir in places where you're reading from or writing to a file, you should use `AzureSession.Instance.DataStore` to perform these operations (wrapper over these `File.*` operations, with a separate implementation for tests)

Reply:
@cormacpayne I have opened the following issue on self - https://github.com/Azure/azure-powershell/issues/9212, to unblock this PR for 5/21 release.